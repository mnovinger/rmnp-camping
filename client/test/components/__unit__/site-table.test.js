import React from 'react';
import renderer from 'react-test-renderer';
import Immutable from 'immutable';
import moment from 'moment';
import SiteTable from '../../../src/app/components/site-table';

describe('The Site Table Component', () => {
  it('should render correctly with site data', () => {
    const props = {
      dates: Immutable.fromJS([moment(), moment().add(1, 'days')]),
      campSiteData: Immutable.fromJS(
        [{
          name: "Boundary Waters",
          id: "001",
          availability : [{
            status: "1"
          }]
        },{
          name: "Olympic",
          id: "002",
          availability: [{
            status: "NA"
          }]
        }
        ]),
    };

    const tree = renderer.create(<SiteTable { ...props }/>).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('should render correctly without site data', () => {
    const props = {
      dates: Immutable.fromJS([moment(), moment().add(1, 'days')]),
      campSiteData: Immutable.Map()
    };

    const tree = renderer.create(<SiteTable { ...props }/>).toJSON();
    expect(tree).toMatchSnapshot();
  })

});