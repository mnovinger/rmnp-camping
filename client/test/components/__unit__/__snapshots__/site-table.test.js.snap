// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`The Site Table Component should render correctly 1`] = `
<div
  className="site-table-container container-fluid"
>
  <table
    className="site-table"
  >
    <thead>
      <tr>
        <th
          className="site-col"
        >
          Site
        </th>
        <th
          className="status-col"
        >
          Fri 03/31
        </th>
        <th
          className="status-col"
        >
          Sat 04/01
        </th>
      </tr>
    </thead>
    <tbody>
      <tr>
        <td
          className="site-col"
        >
          Boundary Waters (001)
        </td>
        <td
          className="status-col"
        >
          1
        </td>
      </tr>
      <tr>
        <td
          className="site-col"
        >
          Olympic (002)
        </td>
        <td
          className="status-col"
        >
          NA
        </td>
      </tr>
    </tbody>
  </table>
</div>
`;
